File: apps/backend/src/backend/models/pokemon.py
from pydantic import BaseModel

class PokemonResponse(BaseModel):
    uid: str
    name: str
    elements: list[str]
    hp: int
    attack: int
    defense: int
    special_attack: int
    special_defense: int
    speed: int
    icon: str
    icon_alt: str
    img: str

File: apps/web/types/pokemon.ts
export type PokemonData = {
    uid: string;
    name: string;
    elements: string[];
    hp: number;
    attack: number;
    defense: number;
    special_attack: number;
    special_defense: number;
    speed: number;
    icon: string;
    icon_alt: string;
    img: string;
}

type Stat = {
    statName: string,
    value: number
}

export type PokemonStats = {
    name: string;
    stats: Stat[]
}